src/gameboard.js

```diff
@@ -1,31 +1,80 @@
-const gameboard = (() => {
+const Gameboard = ((red, black) => {
   const board = [];
+  const emptySquare = { hasMarker: false };
```

Comment:
i agree, i think the object should be `color:` red/black (or just `isRed:`, and black is always false) and `isKing:` true/false.  `hasMarker` itself is not needed, because the board could either contain this piece, or return null (rather than have the object in all spaces, have `hasMarker` true or false on every space, including on all the spots on the gameboard where pieces can never land (all the cream colored tiles).